subsys/shell/backends/shell_mqtt.c

```diff
@@ -0,0 +1,824 @@
+	sh_mqtt->mqtt_cli.client_id.utf8 = (uint8_t *)sh_mqtt->device_id;
+	sh_mqtt->mqtt_cli.client_id.size = strlen(sh_mqtt->device_id);
+	sh_mqtt->mqtt_cli.password = &password;
+	sh_mqtt->mqtt_cli.user_name = &username;
+	sh_mqtt->mqtt_cli.protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	sh_mqtt->mqtt_cli.rx_buf = sh_mqtt->buf.rx;
+	sh_mqtt->mqtt_cli.rx_buf_size = sizeof(sh_mqtt->buf.rx);
+	sh_mqtt->mqtt_cli.tx_buf = sh_mqtt->buf.tx;
+	sh_mqtt->mqtt_cli.tx_buf_size = sizeof(sh_mqtt->buf.tx);
+
+	/* MQTT transport configuration */
+	sh_mqtt->mqtt_cli.transport.type = MQTT_TRANSPORT_NON_SECURE;
+}
+
+/* Work routine to process MQTT packet and keep alive MQTT connection */
+static void sh_mqtt_process_handler(struct k_work *work)
+{
+	ARG_UNUSED(work);
+	int rc;
+	int64_t remaining = LISTEN_TIMEOUT_MS;
+	int64_t start_time = k_uptime_get();
+
+	if (sh_mqtt->network_state != SHELL_MQTT_NETWORK_CONNECTED) {
+		LOG_DBG("%s_work while %s", "process", "network disconnected");
+		return;
+	}
+
+	/* If context can't be locked, that means net conn cb locked it */
+	if (sh_mqtt_context_lock(K_NO_WAIT)) {
+		/* In that case we should simply return */
+		LOG_DBG("%s_work unable to lock context", "process");
+		return;
+	}
+
+	if (sh_mqtt->transport_state != SHELL_MQTT_TRANSPORT_CONNECTED) {
+		LOG_DBG("MQTT %s", "not connected");
+		goto process_error;
+	}
+
+	if (sh_mqtt->subscribe_state != SHELL_MQTT_SUBSCRIBED) {
+		LOG_DBG("%s_work while %s", "process", "MQTT not subscribed");
+		goto process_error;
+	}
+
+	LOG_DBG("MQTT %s", "Processing");
+	/* Listen to the port for a duration defined by LISTEN_TIMEOUT_MS */
+	while ((remaining > 0) && (sh_mqtt->network_state == SHELL_MQTT_NETWORK_CONNECTED) &&
+	       (sh_mqtt->transport_state == SHELL_MQTT_TRANSPORT_CONNECTED) &&
+	       (sh_mqtt->subscribe_state == SHELL_MQTT_SUBSCRIBED)) {
+		LOG_DBG("Listening to socket");
+		if (wait(remaining)) {
+			LOG_DBG("Process socket for MQTT packet");
+			rc = mqtt_input(&sh_mqtt->mqtt_cli);
+			if (rc != 0) {
+				LOG_ERR("%s error: %d", "processed: mqtt_input", rc);
+				goto process_error;
+			}
+		}
+
+		LOG_DBG("MQTT %s", "Keepalive");
+		rc = mqtt_live(&sh_mqtt->mqtt_cli);
+		if (rc != 0 && rc != -EAGAIN) {
+			LOG_ERR("%s error: %d", "mqtt_live", rc);
+			goto process_error;
+		}
+
+		remaining = LISTEN_TIMEOUT_MS + start_time - k_uptime_get();
+	}
+
+	/* Reschedule the process work */
+	LOG_DBG("Scheduling %s work", "process");
+	(void)sh_mqtt_work_reschedule(&sh_mqtt->process_dwork, K_SECONDS(2));
+	sh_mqtt_context_unlock();
+	return;
+
+process_error:
+	LOG_DBG("%s: close MQTT, cleanup socket & reconnect", "connect");
+	sh_mqtt_close_and_cleanup();
+	(void)sh_mqtt_work_reschedule(&sh_mqtt->connect_dwork, K_SECONDS(1));
+	sh_mqtt_context_unlock();
+}
+
+static void sh_mqtt_subscribe_handler(struct k_work *work)
+{
+	ARG_UNUSED(work);
+	/* Subscribe config information */
+	/* clang-format off */
```

Comment:
Why is it needed to disable clang-format here?

Reply:
It looks like this after clang-format:
```c
	struct mqtt_topic subs_topic = { .topic = { .utf8 = sh_mqtt->sub_topic,
						    .size = strlen(sh_mqtt->sub_topic) },
					 .qos = MQTT_QOS_1_AT_LEAST_ONCE };
	const struct mqtt_subscription_list subs_list = { .list = &subs_topic,
							  .list_count = 1U,
							  .message_id = 1U };
```